src/bregman.jl

```diff
@@ -145,14 +145,14 @@ function bregman(funobj::Function, x::AbstractVector{T}, options::BregmanParams=
         @. z = z + d
         # Get λ at first iteration
         (i == 1) && (sol.λ = abs.(T.(options.λfunc(z))))
-        # Save curent state
+        # Save current state
         options.spg && (gold .= g; xold .= x)
         # Update x
         x = options.TD'*soft_thresholding(z, sol.λ)
-
         obj_fun = norm(sol.λ .* z, 1) + .5 * norm(z, 2)^2
+        # Output Log
         (options.verbose > 0) && (@printf("%10d %15.5e %15.5e %15.5e %15.5e \n",i, t, obj_fun, f, maximum(sol.λ)))
-        norm(x - sol.x) < options.progTol && (@printf("Step size below progTol\n"); break;)
+        norm(z - sol.z) < options.progTol && (@printf("Step size below progTol\n"); break;)
```

Comment:
Fix this to check the progress on `z` when `lambda` is too large so that nothing passes in the first iteration

Reply:
No z is the dual bregman solves for x and x should be the one to be checked.

Reply:
But if the threshold is set very high so that nothing pass the threshold in the first iteration, the algorithm will be stopped which is not what we want

Reply:
If the user chooses a threshold too high it's their problem, the default one cannot lead to no progress. That's not a valid convergence check, the optimization and solution is for `x` and is the one the convergence check is valid for.